### PR TITLE
Add support to automatically load Codemirror language mode

### LINKED ISSUE #5072: Add support for automagically loading required CodeMirror mode for languages
**Is your feature request related to a problem? Please describe.**

I have being using jupyter-notebook with in combination with [rust kernel](https://github.com/google/evcxr/tree/master/evcxr_jupyter) and after discovering nteract I tried to use it there.

Good news is it worked out of the box and I did not had to do anything to make it work.

Bad news is (unlike junyper-notebook) code blocks don't seem to be syntax highlighted.

**Describe the solution you'd like**

**Describe alternatives you've considered**
Continue using junyper-notebook

**Additional context**
image
image

### PR DESCRIPTION
**Pull request by bolariin**
- [x] I have read the [Contributor Guide](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md)

This is intended to resolve #5072.  I tested this manually and it looks to be working.

### CODE REVIEW

File: packages/editor/src/index.tsx [hunk 1]
```diff
@@ -350,7 +370,7 @@ export default class CodeMirrorEditor extends React.Component<
         this.props.codeMirror[optionName] !== prevProps.codeMirror[optionName]
       ) {
         if (optionName === "mode") {
-          this.cm.setOption(optionName, this.cleanMode());
+          this.loadMode().then(mode => this.cm.setOption("mode", mode));
```
Comment (captainsafia) [inline packages/editor/src/index.tsx]: Nit: Let's add a catchError here to log a warning to the console in the event that the `loadMode` fails.

**Review by captainsafia — APPROVED** [review]
Looks good overall. Left one comment inline.

What were the issues you were having with this mentioned in the Slack?

**Comment by bolariin** [comment]
> What were the issues you were having with this mentioned in the Slack?

I was running into `Critical dependency: the request of a dependency is an expression` when I had initially used the import statement to load in the language mode. 
It was fixed by changing from 
```js
const modeURL = "codemirror/mode/%N/%N.js".replace(/%N/g, mode);
import(modeURL)
```
to
```js
import(`codemirror/mode/${mode}/${mode}.js`)
```

**Comment by captainsafia** [comment]
I'll merge this now so that we can include it in the June release. Thanks for taking a stab at this and welcome to the nteract party!

**Pull request merged**